espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Uiautomator.java

```
@@ -0,0 +1,50 @@
+package io.appium.espressoserver.lib.handlers;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.BySelector;
+import androidx.test.uiautomator.UiObject2;
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.helpers.ReflectionUtils;
+import io.appium.espressoserver.lib.model.UiautomatorParams;
+
+import static io.appium.espressoserver.lib.helpers.AndroidLogger.logger;
+import static io.appium.espressoserver.lib.helpers.InteractionHelper.getUiDevice;
+
+public class Uiautomator implements RequestHandler<UiautomatorParams, Object> {
+
+    @Override
+    public Object handle(UiautomatorParams params) throws AppiumException {
+        logger.info("Invoking Uiautomator Methods");
+
+        ArrayList<Object> result = new ArrayList<>();
+        String byMethodName = params.getBy();
+        String value = params.getValue();
+        String action = params.getAction();
+        Integer index = params.getIndex();
+
+        try {
+            Method byMethod = ReflectionUtils.method(By.class, byMethodName, String.class);
+            BySelector selector = (BySelector) ReflectionUtils.invoke(byMethod, By.class, value);
+            List<UiObject2> uiObjects = getUiDevice().findObjects(selector);
+
+            Method uiObjectMethod = ReflectionUtils.method(UiObject2.class, action);
```

Some logging here about how many objects were matched would be useful

---

We'd like to fail fast, so it would be more logical to to move method extraction before objects lookup, since this operation is less expensive